internal/userutils/userutils.go

```diff
@@ -0,0 +1,39 @@
+// Package userutils provides functions related to system users and groups.
+package userutils
+
+import (
+	"errors"
+	"fmt"
+	"os"
+)
+
+var groupFile = "/etc/group"
+
+// LockGroupFile creates a lock file at /etc/group.lock, the same location
+// used by tools like gpasswd and groupmod to prevent concurrent modifications
+// to the /etc/group file.
+// The lock file contains the PID of the process that created it.
+func LockGroupFile() (err error) {
+	lockFile, err := os.OpenFile(groupFile+".lock", os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0600)
```

Comment:
Would be possible to use both locking systems as gpasswd (so also `lckpwdf`)?

Reply:
Now that we have a working test which checks that the locking works and gpasswd refuses to modify the file, isn't that good enough? What could we possible gain from using `lckpwdf` as well?

Reply:
Leaving it here for the records: `lckpwdf` is used everywhere while systemd ignores the per files-databases.

https://github.com/systemd/systemd/blob/bf478dcffbc3c568fb3a3c8e8b93b3e7b66f4228/src/basic/user-util.c#L670-L675

And other than that the behavior is different, so let's just go the libc-way.